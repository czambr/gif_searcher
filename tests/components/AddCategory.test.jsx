import { fireEvent, render, screen } from '@testing-library/react';
import { AddCategory } from '../../src/components/AddCategory';

describe('Evaluando el componente <AddCategory />', () => {
    test('Debe hacer Match con el snapshot', () => {
        const { container } = render(<AddCategory onNewCategory={() => {}} />);
        expect(container).toMatchSnapshot();
    });

    test('Debe cambiar el valor de la caja de texto', () => {
        render(<AddCategory onNewCategory={() => {}} />);

        const input = screen.getByRole('textbox');
        fireEvent.input(input, { target: { value: 'goku' } });

        expect(input.value).toBe('goku');
    });

    test('Debe llamar onNewcategory si el input tiene un valor', () => {
        const inputValue = 'Dragon Ball';
        const onNewCategory = jest.fn();

        render(<AddCategory onNewCategory={onNewCategory} />);

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, { target: { value: inputValue } });
        fireEvent.submit(form);

        expect(input.value).toBe('');
        expect(onNewCategory).toHaveBeenCalledTimes(1);
        expect(onNewCategory).toHaveBeenCalledWith(inputValue);
    });

    test('No debe de llamar el onNewCategory si el input estÃ¡ vacio', () => {
        const onNewCategory = jest.fn();
        render(<AddCategory onNewCategory={onNewCategory} />);

        const form = screen.getByRole('form');
        fireEvent.submit(form);

        expect(onNewCategory).not.toHaveBeenCalled();
    });
});
