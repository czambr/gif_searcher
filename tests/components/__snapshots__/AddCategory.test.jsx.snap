// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Evaluando el componente <AddCategory /> Debe hacer Match con el snapshot 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
